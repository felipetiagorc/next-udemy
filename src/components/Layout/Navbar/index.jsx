import {
  Avatar,
  Divider,
  Flex,
  Heading,
  IconButton,
  Text
} from '@chakra-ui/react'
import { useState } from 'react'
import { FiCalendar, FiHome, FiMenu, FiSettings, FiUser } from 'react-icons/fi'
import NavItem from './NavItem'

export default function NavBar() {
  const [navSize, setNavSize] = useState('large')
  return (
    <Flex
      pos="sticky"
      left="5"
      h="95vh"
      marginTop="2.5vh"
      boxShadow="0 4px 12px 0 rgba(0,0,0,0.05)"
      borderRadius={navSize == 'small' ? '15px' : '30px'}
      w={navSize == 'small' ? '75px' : '200px'}
      flexDir="column"
      justifyContent="space-between"
      backgroundColor="red.200"
    >
      <Flex p="5%" flexDir="column" alignItems="flex-start" as="nav">
        <IconButton
          background="none"
          mt={5}
          _hover={{ background: 'none' }}
          aria-label="botao"
          icon={<FiMenu />}
          onClick={() => {
            if (navSize == 'small') setNavSize('large')
            else setNavSize('small')
          }}
        />
        <NavItem navSize={navSize} icon={FiHome} title={'DashBoard'} />
        <NavItem navSize={navSize} icon={FiCalendar} title={'Calendar'} />
        <NavItem navSize={navSize} icon={FiUser} title={'User'} />
        <NavItem navSize={navSize} icon={FiSettings} title={'Settings'} />
      </Flex>
      <Flex
        p="5%"
        flexDir="column"
        w="100%"
        alignItems={navSize == 'small' ? 'center' : 'flex-start'}
        mb={4}
      >
        <Divider display={navSize == 'small' ? 'none' : 'flex'} />
        <Flex mt="4" align="center">
          <Avatar size="sm" src="" />
          <Flex
            flexDirection="column"
            ml="4"
            display={navSize == 'small' ? 'none' : 'flex'}
          >
            <Heading as="h3" size="sm">
              Silvio
            </Heading>
            <Text>Admin</Text>
          </Flex>
        </Flex>
      </Flex>
    </Flex>
  )
}
