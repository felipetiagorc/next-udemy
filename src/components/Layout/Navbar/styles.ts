<<<<<<< HEAD
import styled, { css } from 'styled-components'

export const SidebarContainer = styled.div`
    flex: 1;
    height: calc(100vh - 50px);
    background-color: rgb(251, 251, 255);
    position: sticky;
    top: 50px;
`
export const SidebarWrapper = styled.div`
    padding: 20px;
    color: #555;
`
export const SidebarMenu = styled.div`
    margin-bottom: 10px;
`
export const SidebarTitle = styled.h3`
    font-size: 13px;
    color: rgb(187, 186, 186);
`
export const SidebarList = styled.ul`
    list-style: none;
    padding: 5px;
`
export const SidebarListItem = styled.li`
    padding: 5px;
    cursor: pointer;
    display: flex;
    align-items: center;
    border-radius: 10px;&:hover {
        background-color: rgb(240, 240, 255);
    }
`
export const sharedStyle = css`
    margin-right: 5px;
    font-size: 20px !important;
    `
export const MyLineStyle = styled(LineStyle)`
    ${sharedStyle}
`
export const MyTimeline = styled(Timeline)`
    ${sharedStyle}
`
export const MyTrendingUp = styled(TrendingUp)`
    ${sharedStyle}
`
export const MyPermIdentity = styled(PermIdentity)`
    ${sharedStyle}
`
export const MyStorefront = styled(Storefront)`
    ${sharedStyle}
`
export const MyAssessment = styled(Assessment)`
    ${sharedStyle}
`
export const MyLocalAtm = styled(LocalAtm)`
    ${sharedStyle}
`
export const MyDrafts = styled(Drafts)`
    ${sharedStyle}
`
export const MyFeedback = styled(Feedback)`
    ${sharedStyle}
`
export const MyForum = styled(Forum)`
    ${sharedStyle}
`
export const MyWork = styled(Work)`
    ${sharedStyle}
`
export const MyPieChart = styled(PieChart)`
    ${sharedStyle}
`
export const MyReceipt = styled(Receipt)`
    ${sharedStyle}
`
=======
import styled from 'styled-components'

export const NavBar = styled.aside`
  position: static;
  background-color: grey;
  margin-left: 0%;
  height: 50vh;
  max-width: 40vh;
  font-size: var(--medium);
  line-height: var(--medium);
`
>>>>>>> cfc9beb386e8f33888c928d81881b69962b0dbc0
