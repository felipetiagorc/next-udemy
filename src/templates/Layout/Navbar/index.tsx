import {
  Avatar,
  Divider,
  Flex,
  Heading,
  IconButton,
  Text
} from '@chakra-ui/react'
import { useState } from 'react'
import { FiMenu } from 'react-icons/fi'

export default function NavBar() {
  const [navSize, setNavSize] = useState('large')
  return (
    <Flex
      pos="sticky"
      left="5"
      h="95vh"
      marginTop="2.5vh"
      boxShadow="0 4px 12px 0 rgba(0,0,0,0.05)"
      w="200px"
      flexDir="column"
      justifyContent="space-between"
    >
      <Flex p="5%" flexDir="column" alignItems="flex-start" as="nav">
        <IconButton
          background="none"
          mt={5}
          _hover={{ background: 'none' }}
          aria-label="botao"
          icon={<FiMenu />}
          onClick={() => {
            if (navSize == 'small') setNavSize('large')
            else setNavSize('small')
          }}
        />
      </Flex>
      <Flex p="5%" flexDir="column" w="100%" alignItems="flex-start" mb="4">
        <Divider />
        <Flex mt="4" align="center">
          <Avatar size="sm" src="" />
          <Flex flexDirection="column" ml="4">
            <Heading as="h3" size="sm">
              Silvio
            </Heading>
            <Text>Admin</Text>
          </Flex>
        </Flex>
      </Flex>
    </Flex>
  )
}
