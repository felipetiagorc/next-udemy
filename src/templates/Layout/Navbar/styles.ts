import styled from 'styled-components'

export const NavBar = styled.aside`
  margin-left: 0%;
  height: 100vh;
  max-width: var(--container);
  margin: auto;
  font-size: var(--medium);
  line-height: var(--medium);
`
