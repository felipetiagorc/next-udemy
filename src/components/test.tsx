import { render, screen } from '@testing-library/react'

import Navbar from './navbar'

describe('<navbar/>', () => {
  it('deve retornar heading', () => {
    const { container } = render(<Navbar />)
    expect(
      screen.getByRole('heading', { name: /PanelMenu/i })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
