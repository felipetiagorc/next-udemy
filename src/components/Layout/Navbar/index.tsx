import React from 'react'
import * as S from './styles'
const Sidebar = () => {
  return (
    <S.SidebarContainer>
      <SidebarWrapper>
        <SidebarMenu>
          <SidebarTitle>Dashboard</SidebarTitle>
          <SidebarList>
            <SidebarListItem>
              <MyLineStyle />
              Home
            </SidebarListItem>
            <SidebarListItem>
              <MyTimeline />
              Analytics
            </SidebarListItem>
            <SidebarListItem>
              <MyTrendingUp />
              Sales
            </SidebarListItem>
          </SidebarList>
        </SidebarMenu>
        <SidebarMenu>
          <SidebarTitle>All Menu</SidebarTitle>
          <SidebarList>
            <SidebarListItem>
              <MyPermIdentity />
              Users
            </SidebarListItem>
            <SidebarListItem>
              <MyStorefront />
              Products
            </SidebarListItem>
            <SidebarListItem>
              <MyAssessment />
              Reports
            </SidebarListItem>
            <SidebarListItem>
              <MyLocalAtm />
              Transactions
            </SidebarListItem>
          </SidebarList>
        </SidebarMenu>
        <SidebarMenu>
          <SidebarTitle>Connect</SidebarTitle>
          <SidebarList>
            <SidebarListItem>
              <MyDrafts />
              Mail
            </SidebarListItem>
            <SidebarListItem>
              <MyFeedback />
              Feedback
            </SidebarListItem>
            <SidebarListItem>
              <MyForum />
              Messages
            </SidebarListItem>
          </SidebarList>
        </SidebarMenu>
        <SidebarMenu>
          <SidebarTitle>Employees</SidebarTitle>
          <SidebarList>
            <SidebarListItem>
              <MyWork />
              Manage
            </SidebarListItem>
            <SidebarListItem>
              <MyPieChart />
              Analytics
            </SidebarListItem>
            <SidebarListItem>
              <MyReceipt />
              Reports
            </SidebarListItem>
          </SidebarList>
        </SidebarMenu>
      </SidebarWrapper>
    </SidebarContainer>
  )
}
export default Sidebar
