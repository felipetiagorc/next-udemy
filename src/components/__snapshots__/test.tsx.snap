// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<navbar/> deve retornar heading 1`] = `
<h2>
  PanelMenu
</h2>
`;
